Automattic/wp-calypso
# Update account deleted page to allow restore behind feature flag

Related to https://github.com/Automattic/dotcom-forge/issues/9460

> [!NOTE]  
> To keep this PR contained as it's already changing 10 files, let's focus on the functional aspects. If you notice any design improvements, please comment in the [original issue](https://github.com/Automattic/dotcom-forge/issues/9460)

## Proposed Changes

Update: I've also updated the deleting state screen to this:

| Before  | After |
| :-------------: | :-------------: |
| <img src="https://github.com/user-attachments/assets/66675f57-079e-44d1-a268-310a4efaf342">  | <img src="https://github.com/user-attachments/assets/1c87f4c7-e9f0-493f-93d6-79c4d5475636">|


Original flow (without restore function)

https://github.com/user-attachments/assets/f35c3721-9ff4-476c-88f8-4c6be247f029

With the feature flag `?flags=me/account-restore`


https://github.com/user-attachments/assets/b8079003-7ac5-41ad-95f0-42efb983588f


## Testing Instructions



Make sure we didn't break anything on production:
* Go to `/me/account/close` and delete the account
* it should be the same as production except we have a new layout for the deleted screen

Enable the feature flag
* Go to /me/account/close?flags=me/account-restore
* Go through the process and check for account restoration

> [!NOTE]  
> Currently after refreshing `/me/account/closed`, it will redirect back to a login screen as we haven't yet allowed that page to be accessed in a logged out state. This will be addressed in https://github.com/Automattic/dotcom-forge/issues/9909

## Pre-merge Checklist
